bcl/src/bc_system.c

```diff
@@ -185,6 +193,27 @@ static void _bc_system_init_gpio(void)
     GPIOA->MODER |= GPIO_MODER_MODE4_1 | GPIO_MODER_MODE4_0;
 }
 
+static void _bc_system_detect_core_version(void)
+{
+    bc_irq_disable();
+
+    // Input mode
+    GPIOA->MODER &= ~GPIO_MODER_MODE11_Msk;
+
+    // Pull Up
+    GPIOA->PUPDR |= GPIO_PUPDR_PUPD11_0;
+
+    _bc_system_core_version = (GPIOA->IDR & GPIO_IDR_ID11) != 0 ? BC_SYSTEM_CORE_VERSION_R1 : BC_SYSTEM_CORE_VERSION_R2;
```

Comment:
There should be some delay between pull-up activation and the input reading. Simply because there is some capacitance of the pin and it takes a little while to charge it. While this may work, I bet it is not a safe solution.